Mlem/Models/Trackers/Feeds/Modlog/ModlogChildTracker.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ModlogChildTracker.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-03-14.
+//
+
+import Dependencies
+import Foundation
+
+/// Class to handle modlog children. Because the API paginates the modlog per-action type, modlog needs to be handled using a multi-tracker; however, because all modlog entries are represented with ModlogEntry, we can define a single generic child tracker instead of needing 14 different ones
+class ModlogChildTracker: ChildTracker<ModlogEntry, ModlogEntry>, ModlogTrackerProtocol {
+    @Dependency(\.apiClient) var apiClient
+    
+    private let actionType: APIModlogActionType
+    private let instance: URL?
+    private let communityId: Int?
+    
+    init(
+        internetSpeed: InternetSpeed,
+        sortType: TrackerSortType,
+        actionType: APIModlogActionType,
+        instance: URL?,
+        communityId: Int?
+    ) {
+        self.actionType = actionType
+        self.instance = instance
+        self.communityId = communityId
+        
+        super.init(internetSpeed: internetSpeed, sortType: sortType)
+    }
+    
+    override func toParent(item: ModlogEntry) -> ModlogEntry {
+        item
+    }
+    
+    init(internetSpeed: InternetSpeed, sortType: TrackerSortType, actionType: APIModlogActionType, modlogLink: ModlogLink) {
+        self.actionType = actionType
+        switch modlogLink {
+        case .userInstance:
+            self.instance = nil
+            self.communityId = nil
+        case let .instance(instance):
+            self.instance = instance
+            self.communityId = nil
+        case let .community(community):
+            self.instance = nil
+            self.communityId = community.communityId
+        }
+        
+        super.init(internetSpeed: internetSpeed, sortType: sortType)
+    }
+    
+    override func fetchPage(page: Int) async throws -> FetchResponse<ModlogEntry> {
+        let items = try await apiClient.getModlog(
```

Comment:
This means making thirteen separate requests for each modlog page that we load - one for each child tracker. Is there a better way to do this? Maybe page loading could be delegated to the parent tracker somehow, so only one request needs to be made per page?

Reply:
The trouble arises from the fact that certain actions are far more frequent--we need to load maybe 20 pages of post removals for every page of community transfers, so loading full API pages from the parent tracker would require loading lots and lots of unused data. This is one of the core problems that child trackers were built to solve--they only load the next *n* items of a particular type when that type runs out, rather than loading everything from every type every time. 

The initial preheat, however, is definitely way more expensive than it needs to be. I'm looking into a way to use a single call to preheat the children and then switch over to ad hoc loading.

Reply:
If the child trackers request the nth page from the parent instead of directly requesting it, the parent could maintain a cache of every page loaded so far and only fetch the next page if it isn't yet cached.

Reply:
Implemented ad-hoc with first page optimizations per Slack